### PR TITLE
added object_it_functions

### PR DESCRIPTION
**Pull request by Jasper-Ben**
returns the number of objects in object iterator

### CODE REVIEW

**Comment by heregoesmarcel** [comment]
Correct me if I'm wrong but both functions seem to be not safe to use if `it` is supposed to keep its given offset after the function call. I'd rather like to see a  copy of `it` performing the `rewind()` and `next()`operations.

**Comment by Jasper-Ben** [comment]
> Correct me if I'm wrong but both functions seem to be not safe to use if `it` is supposed to keep its given offset after the function call. I'd rather like see a copy of `it` performing the `rewind()` and `next()`operations.

This indeed might be an issue, I myself have been thinking about that. If this is the case, the same issue will occur in the [array_it_length](https://codedocs.xyz/jaksonlabs/jakson/array__it_8c_source.html#l00199) function and we should fix that. I've opened a separate issue at [https://github.com/jaksonlabs/jakson/issues/2](https://github.com/jaksonlabs/jakson/issues/2).

**Comment by Jasper-Ben** [comment]
also added a function to get raw data with a given key.

File: src/jakson/carbon/object_it.c [hunk 1]
```diff
@@ -149,6 +164,24 @@ bool carbon_object_it_prev(carbon_object_it *it)
...
+bool carbon_object_it_has_key(const char *key, carbon_object_it *it) {
+
+    ERROR_IF_NULL(key);
+    ERROR_IF_NULL(it);
+
+    carbon_object_it_rewind(it);
+    while (carbon_object_it_next(it)) {
+        u64 namelen;
+        const char *_name = carbon_object_it_prop_name(&namelen, it);
+        const char *name = strndup(_name, namelen);
```
Comment (Jasper-Ben) [inline src/jakson/carbon/object_it.c]: memleak, fixme

File: src/jakson/carbon/object_it.c [hunk 2]
```diff
@@ -325,6 +358,31 @@ carbon_column_it *carbon_object_it_column_value(carbon_object_it *it_in)
...
+    ERROR_IF_NULL(data);
+    ERROR_IF_NULL(key);
+    ERROR_IF_NULL(it);
+
+    carbon_object_it_rewind(it);
+  
+    while (carbon_object_it_has_next(it)) {
+        u64 keylen;
+        const char *_comp = carbon_object_it_prop_name(&keylen, it);
+        const char *comp = strndup(_comp, keylen);
```
Comment (Jasper-Ben) [inline src/jakson/carbon/object_it.c]: memleak, fixme

File: src/jakson/carbon/object_it.c [hunk 3]
```diff
@@ -358,6 +358,31 @@ carbon_column_it *carbon_object_it_column_value(carbon_object_it *it_in)
...
+
+        if (!(strcmp(key, comp))) {
+            data = &(it->field.value.data.it_field_data);
```
Comment (Jasper-Ben) [inline src/jakson/carbon/object_it.c]: change to `data = &(it->field.value.data);`

**Pull request closed without merging**